hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/IncrSourceHelper.java

```diff
@@ -130,7 +130,7 @@ public static QueryInfo generateQueryInfo(JavaSparkContext jssc, String srcBaseP
       }
     });
 
-    String previousInstantTime = beginInstantTime;
+    String previousInstantTime = DEFAULT_BEGIN_TIMESTAMP;
     if (!beginInstantTime.equals(DEFAULT_BEGIN_TIMESTAMP)) {
       Option<HoodieInstant> previousInstant = activeCommitTimeline.findInstantBefore(beginInstantTime);
       if (previousInstant.isPresent()) {
```

Comment:
`beginInstantTime == 000` case would already be handled by current implementation 
`String previousInstantTime = beginInstantTime;`.

It seems the problem is because of if statement here? There are two cases I could think of where  `previousInstantTime` is not set to correct previous instant :-
1. `beginInstantTime` is the first commit in activeTimeline. In such a case we should also include archived timeline for polling the instant before.
2. `beginInstantTime` is the first commit in archivedTimeline. In this case we will need to set it to `DEFAULT_BEGIN_TIMESTAMP`.

Reply:
This is not a valid scenario to query using archival timeline

Reply:
synced up w/ lokesk J.guess this is what he is saying. 
lets say active timeline is from C5 until C10. 
when beginInstantTime is C5, as per this patch, previous will be set to 00000000000000. and down the line, when we trigger incremental query, we will set previous i.e. 00000000000000) as the begin instant time which is not what we want.